src/System.Management.Automation/help/HelpProvider.cs

```diff
@@ -223,10 +223,7 @@ internal void ReportHelpFileError(Exception exception, string target, string hel
         /// <returns>String representing base directory of the executing shell.</returns>
         internal string GetDefaultShellSearchPath()
         {
-            string shellID = this.HelpSystem.ExecutionContext.ShellID;
-            // Beginning in PowerShell 6.0.0.12, the $pshome is no longer registry specified, we search the application base instead.
-            // We use executing assemblies location in case registry entry not found
-            return Utils.GetApplicationBase() ?? Path.GetDirectoryName(Environment.ProcessPath);
```

Comment:
> Path.GetDirectoryName(Environment.ProcessPath)
I meant that we should use this fallback as last resort.

Reply:
fixed